### ISSUE TITLE
Subject Alt Name not being honoured

### ISSUE THREAD

**Issue by cbluth** [rancher/k3s#469]
**Describe the bug**
`curl -sfL https://get.k3s.io | sh -s - server --cluster-cidr 10.235.32.0/19 --service-cidr 10.235.0.0/19 --tls-san k8s.example.com --bind-address 0.0.0.0`

^ install like so, edit KUBECONFIG to use desired hostname.

```
$ kubectl --context=dev get node
Unable to connect to the server: x509: certificate is valid for kubernetes.default.svc, kubernetes.default, kubernetes, localhost, master-node, not k8s.example.com
```
^ then this happens

hostname validation fails.

**To Reproduce**
see above

**Expected behavior**
kubectl to work without hostname validation issues

**Comment by cbluth** [comment]
```
root@master-ao7vs:~# openssl x509 -text -in /var/lib/rancher/k3s/server/tls/localhost.crt 
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number: 4763654641946726837 (0x421be6a86de011b5)
    Signature Algorithm: sha256WithRSAEncryption
        Issuer: CN=k3s-tls-ca@1557506098
        Validity
            Not Before: May 10 16:34:58 2019 GMT
            Not After : May  9 16:34:58 2020 GMT
        Subject: CN=localhost
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                Public-Key: (2048 bit)
                Modulus:
                    00:c4:cd:6b:db:cf:31:62:35:e2:9c:a0:8f:70:7d:
                    c8:33:16:87:1b:8f:15:d2:4b:9b:75:4d:35:49:71:
                    67:95:a5:20:2b:4f:9f:07:6b:86:8c:70:b9:9a:fd:
                    a3:c4:c3:b4:8d:e7:12:cb:08:2f:54:9c:47:3c:ca:
                    6f:6b:e9:14:48:a5:03:83:76:b4:5e:ce:f3:c8:56:
                    63:47:8f:df:a2:1d:61:80:c3:7e:d0:98:7a:91:47:
                    52:c0:1e:46:1a:42:f5:c1:22:f4:a4:f7:4c:3d:25:
                    37:93:db:e3:b0:ba:fb:45:03:11:7e:06:d0:95:61:
                    ec:9d:67:10:e3:c2:e8:65:47:aa:98:58:e3:c1:22:
                    58:8e:bf:b2:96:0e:ff:60:77:2a:88:0a:46:16:0c:
                    dd:4d:39:51:8c:74:ff:1a:88:7b:89:2e:29:4d:e6:
                    aa:32:47:09:78:4d:03:8f:2f:1e:aa:37:d7:a7:62:
                    da:0b:3b:03:16:bb:df:fa:60:bd:62:3d:60:79:5c:
                    d0:81:b8:a1:da:e7:55:f9:cb:f6:4e:fe:cb:23:85:
                    76:9c:d6:4c:19:42:d2:88:2a:77:44:aa:b9:36:ec:
                    ab:f7:30:76:21:cf:64:21:d2:24:a1:8e:4e:67:18:
                    91:34:1e:57:a0:93:e1:03:3d:bc:b7:02:65:bf:eb:
                    40:cb
                Exponent: 65537 (0x10001)
        X509v3 extensions:
            X509v3 Key Usage: critical
                Digital Signature, Key Encipherment
            X509v3 Extended Key Usage: 
                TLS Web Server Authentication
            X509v3 Subject Alternative Name: 
                DNS:kubernetes.default.svc, DNS:kubernetes.default, DNS:kubernetes, DNS:localhost, IP Address:10.235.0.1, IP Address:127.0.0.1
    Signature Algorithm: sha256WithRSAEncryption
         22:01:56:d7:72:44:f3:a7:4d:50:ce:07:c8:79:9a:31:2d:79:
         f5:8b:e8:ef:56:6d:e9:f1:89:38:b1:a8:26:b0:80:af:f7:32:
         07:46:4b:78:36:48:4b:04:47:0d:43:a5:69:93:21:4f:92:c2:
         5c:0f:37:e9:16:e8:a2:10:0f:a6:61:04:da:b3:18:2b:c3:46:
         49:2a:44:03:3e:bd:fc:0f:ca:43:a2:70:af:4f:90:dd:33:57:
         32:4a:d2:fc:5e:c9:8e:8c:e3:5a:0e:16:21:b8:b0:88:e8:8b:
         6c:14:85:31:cf:ee:6d:60:35:c0:aa:3a:3f:70:87:e7:30:a0:
         ef:76:43:11:a3:34:3d:19:5e:06:b7:5c:dc:f8:83:11:f5:97:
         04:c6:18:9b:39:24:24:a6:ed:4e:ea:84:c5:9a:b9:d0:0f:45:
         2b:43:40:97:48:fb:d3:d3:41:02:05:48:9f:12:59:c9:70:b3:
         b1:17:40:ad:86:20:9f:49:cb:4e:b6:af:03:69:86:c0:f3:a4:
         b2:40:db:a3:23:2b:cd:cd:2d:51:2d:af:d8:25:3d:8d:c8:a9:
         79:69:18:28:62:36:51:0c:28:50:3a:6a:d5:9b:30:11:e3:67:
         25:40:c1:2c:cb:05:6b:4a:b3:38:05:e3:5a:38:40:29:41:e4:
         72:9d:0f:0d
-----BEGIN CERTIFICATE-----
MIIDODCCAiCgAwIBAgIIQhvmqG3gEbUwDQYJKoZIhvcNAQELBQAwIDEeMBwGA1UE
AwwVazNzLXRscy1jYUAxNTU3NTA2MDk4MB4XDTE5MDUxMDE2MzQ1OFoXDTIwMDUw
OTE2MzQ1OFowFDESMBAGA1UEAxMJbG9jYWxob3N0MIIBIjANBgkqhkiG9w0BAQEF
AAOCAQ8AMIIBCgKCAQEAxM1r288xYjXinKCPcH3IMxaHG48V0kubdU01SXFnlaUg
K0+fB2uGjHC5mv2jxMO0jecSywgvVJxHPMpva+kUSKUDg3a0Xs7zyFZjR4/foh1h
gMN+0Jh6kUdSwB5GGkL1wSL0pPdMPSU3k9vjsLr7RQMRfgbQlWHsnWcQ48LoZUeq
mFjjwSJYjr+ylg7/YHcqiApGFgzdTTlRjHT/Goh7iS4pTeaqMkcJeE0Djy8eqjfX
p2LaCzsDFrvf+mC9Yj1geVzQgbih2udV+cv2Tv7LI4V2nNZMGULSiCp3RKq5Nuyr
9zB2Ic9kIdIkoY5OZxiRNB5XoJPhAz28twJlv+tAywIDAQABo4GBMH8wDgYDVR0P
AQH/BAQDAgWgMBMGA1UdJQQMMAoGCCsGAQUFBwMBMFgGA1UdEQRRME+CFmt1YmVy
bmV0ZXMuZGVmYXVsdC5zdmOCEmt1YmVybmV0ZXMuZGVmYXVsdIIKa3ViZXJuZXRl
c4IJbG9jYWxob3N0hwQK6wABhwR/AAABMA0GCSqGSIb3DQEBCwUAA4IBAQAiAVbX
ckTzp01QzgfIeZoxLXn1i+jvVm3p8Yk4sagmsICv9zIHRkt4NkhLBEcNQ6VpkyFP
ksJcDzfpFuiiEA+mYQTasxgrw0ZJKkQDPr38D8pDonCvT5DdM1cyStL8XsmOjONa
DhYhuLCI6ItsFIUxz+5tYDXAqjo/cIfnMKDvdkMRozQ9GV4Gt1zc+IMR9ZcExhib
OSQkpu1O6oTFmrnQD0UrQ0CXSPvT00ECBUifElnJcLOxF0CthiCfSctOtq8DaYbA
86SyQNujIyvNzS1RLa/YJT2NyKl5aRgoYjZRDChQOmrVmzAR42clQMEsywVrSrM4
BeNaOEApQeRynQ8N
-----END CERTIFICATE-----
root@master-ao7vs:~#
```

the subject alt name is not being added

**Comment by iMartyn** [comment]
Is there a workaround for this one? i.e. is the ca key in an obvious place where we can re-issue the apiserver cert manually?

**Comment by erikwilson** [comment]
Fixed with https://github.com/rancher/k3s/pull/359/commits/2c9444399b427ffb706818f5bf3892a8880673bf#diff-03d02d29b94fb1b982922f746ce5dcb0R106

**Comment by erikwilson** [comment]
Fixed with v0.7.0-rc9, used `--tls-san k8s.example.com` flag and verified record exists with `echo | openssl s_client -showcerts -connect localhost:6443  2>/dev/null | openssl x509 -text | grep DNS`
